pandas/tseries/period.py

```diff
@@ -271,9 +271,14 @@ def _from_arraylike(cls, data, freq, tz):
 
     @classmethod
     def _simple_new(cls, values, name=None, freq=None, **kwargs):
-        if not getattr(values, 'dtype', None):
-            values = np.array(values, copy=False)
-        if is_object_dtype(values):
+
+        if lib.is_integer_array(np.array(values, copy=False)):
+            values = np.array(values, copy=False, dtype='int64')
+        elif (len(values) > 0 and
+              lib.is_float_array(np.array(values, copy=False))):
```

Comment:
am fine with failing floats here. 
So you *may* need to consolidate checks that are currently in ``__new__`` (and do them here).